core/src/banking_stage/consumer.rs

```diff
@@ -1065,6 +1092,164 @@ mod tests {
         Blockstore::destroy(ledger_path.path()).unwrap();
     }
 
+    #[test]
+    fn test_bank_process_and_record_transactions_in_last_tick() {
+        solana_logger::setup();
+        let GenesisConfigInfo {
+            genesis_config,
+            mint_keypair,
+            ..
+        } = create_slow_genesis_config(10_000);
+        let bank = Bank::new_no_wallclock_throttle_for_tests(&genesis_config).0;
+        let pubkey = solana_sdk::pubkey::new_rand();
+
+        // setup nonce account with a durable nonce different from the current
+        // bank so that it can be advanced in this bank
+        let durable_nonce = nonce::state::DurableNonce::from_blockhash(&Hash::new_unique());
+        let nonce_hash = *durable_nonce.as_hash();
+        let nonce_pubkey = solana_sdk::pubkey::new_rand();
+        let nonce_state = nonce::State::Initialized(nonce::state::Data {
+            authority: mint_keypair.pubkey(),
+            durable_nonce,
+            fee_calculator: FeeCalculator::new(5000),
+        });
+
+        store_nonce_account(&bank, nonce_pubkey, nonce_state);
+
+        // setup a valid nonce tx which will fail during execution
+        let transactions = sanitize_transactions(vec![system_transaction::nonced_transfer(
+            &mint_keypair,
+            &pubkey,
+            u64::MAX,
+            &nonce_pubkey,
+            &mint_keypair,
+            nonce_hash,
+        )]);
+
+        let ledger_path = get_tmp_ledger_path_auto_delete!();
+        {
+            let blockstore = Blockstore::open(ledger_path.path())
+                .expect("Expected to be able to open database ledger");
+            let (poh_recorder, entry_receiver, record_receiver) = PohRecorder::new(
+                bank.tick_height(),
+                bank.last_blockhash(),
+                bank.clone(),
+                Some((4, 4)),
+                bank.ticks_per_slot(),
+                &pubkey,
+                Arc::new(blockstore),
+                &Arc::new(LeaderScheduleCache::new_from_bank(&bank)),
+                &PohConfig::default(),
+                Arc::new(AtomicBool::default()),
+            );
+            let recorder = poh_recorder.new_recorder();
+            let poh_recorder = Arc::new(RwLock::new(poh_recorder));
+
+            fn poh_tick_before_returning_record_response(
+                record_receiver: Receiver<Record>,
+                poh_recorder: &Arc<RwLock<PohRecorder>>,
+            ) -> JoinHandle<()> {
+                let poh_recorder = poh_recorder.clone();
```

Comment:
*minor*
For `Arc`s, if the method needs to own an instance, it is customary to require the caller to do the clone, I think.
This allows the caller to pass the ownership, if they do not need a copy.

In other words, `Arc`s should not be passed by reference, except if the function clones the object conditionally, and you want to optimize the operation.

```suggestion
                poh_recorder: Arc<RwLock<PohRecorder>>,
            ) -> JoinHandle<()> {
```